bundler/spec/bundler/dsl_spec.rb

```diff
@@ -95,6 +95,33 @@
       expect { subject.eval_gemfile("Gemfile") }.
         to raise_error(Bundler::GemfileError, /There was an error evaluating `Gemfile`: ruby_version must match the :engine_version for MRI/)
     end
+
+    it "raises a warning when no global source has been defined" do
```

Comment:
Let me know what you think about tests. Are these enough? 
I've also tried to follow the same approach in other tests here, but let me know if it's better to change anything.